src/mon/MDSMonitor.cc

```diff
@@ -1459,9 +1459,9 @@ int MDSMonitor::filesystem_command(
 
     bool modified = fsmap.undamaged(role.fscid, role.rank);
     if (modified) {
-      dout(1) << "repaired: restoring rank " << role << dendl;
+      ss << "repaired: restoring rank " << role;
     } else {
-      dout(1) << "repaired: no-op on rank " << role << dendl;
+      ss << "repaired: no-op on rank " << role;
```

Comment:
If both this is essentially a no-op, it's basically the same as saying it was modified from the user's point of view. I don't think it's useful to return to the user the fact that it is a no-op, even if it's useful to log it.

I'd just push the return message to `ss` regardless of the branch, and log each branch's message to dout nonetheless.

Reply:
> If both this is essentially a no-op, it's basically the same as saying it was modified from the user's point of view. I don't think it's useful to return to the user the fact that it is a no-op, even if it's useful to log it.

These messages across the MDSMonitor tend to err on the side of overly helpful (or absent :frowning:). I think it's okay to give helpful information (that the rank is not damaged).

With that said, "no-op" is not the right message, let's use:

`ss << "nothing to do: rank is not damaged";`

> I'd just push the return message to ss regardless of the branch, and log each branch's message to dout nonetheless.

We don't do that for any other message so I don't see a reason to start. These commands are already logged.

Reply:
Yeah, that's fine with me. The 'no-op' was the bit that really seemed off; especially because no-ops in the monitor tend to be assumed as an idempotent operation of whatever we're trying to achieve. If not being damaged is the information we want actually want to relay, then :+1: